From 69ea4773c0aa109747f58f744a68142f11c7c46c Mon Sep 17 00:00:00 2001
From: Lukas Hejtmanek <xhejtman@ics.muni.cz>
Date: Wed, 19 Aug 2009 12:31:55 -0400
Subject: [PATCH] librpcsecgss: be more verbose if context creation fails

the following little patch adds more verbosity to librpcsecgss if context
creation fails. In some cases, rpc.gssd just silently fails to create GSS
context and debuging on server is needed. With this patch, the error codes
are logged.

Signed-off-by: Lukas Hejtmanek <xhejtman@ics.muni.cz>
Signed-off-by: Kevin Coffman <kwc@citi.umich.edu>
---
 src/auth_gss.c |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/src/auth_gss.c b/src/auth_gss.c
index 12e8b75..bbf66fc 100644
--- a/src/auth_gss.c
+++ b/src/auth_gss.c
@@ -524,8 +524,11 @@ authgss_refresh(AUTH *auth)
 
 			if (call_stat != RPC_SUCCESS ||
 			    (gr.gr_major != GSS_S_COMPLETE &&
-			     gr.gr_major != GSS_S_CONTINUE_NEEDED))
+			     gr.gr_major != GSS_S_CONTINUE_NEEDED)) {
+				authgss_log_status("clnt_call", gr.gr_major,
+						    gr.gr_minor);
 				return FALSE;
+			}
 
 			if (gr.gr_ctx.length != 0) {
 				if (gd->gc.gc_ctx.value)
@@ -587,6 +590,8 @@ authgss_refresh(AUTH *auth)
 		authgss_destroy(auth);
 		auth = NULL;
 		rpc_createerr.cf_stat = RPC_AUTHERROR;
+		authgss_log_status("End context negotiation loop",
+				    maj_stat, min_stat);
 
 		return (FALSE);
 	}
-- 
1.7.4.1

